### PR TITLE
eks-prow-build-cluster: Setting up Boskos

### PR DESCRIPTION
**Pull request by pkprzekwas**
These are initial manifests for setting up Boskos instance on our EKS cluster.

ref https://github.com/kubernetes/k8s.io/issues/5048

assign: @xmudrii

### CODE REVIEW

File: infra/aws/terraform/prow-build-cluster/resources/boskos/config.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+metadata:
+  creationTimestamp: null
+  name: resources
+  namespace: test-pods
+data:
+  config: |
+    resources:
+    - names:
+        - k8s-aws-boskos-01
+      state: free
```
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/resources/boskos/config.yaml:25]: I think that this should be dirty. cc @dims

Comment (dims) [reply infra/aws/terraform/prow-build-cluster/resources/boskos/config.yaml:25]: i think so as well! https://cs.k8s.io/?q=state%3A%20(dirty%7Cfree)&i=nope&files=&excludeFiles=&repos=kubernetes/k8s.io,kubernetes/test-infra

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**